### PR TITLE
favGifSearch: don't error on favourited non-urls

### PR DESCRIPTION
**Pull request by Ratismal**
It's possible to save any arbitrary string of text inside of the favourited gif picker, instead of just URLs. This causes favGifSearch to fail with this error:
```
Uncaught TypeError: Failed to construct 'URL': Invalid URL
    at Sb (index.tsx:203:17)
    at index.tsx:163:67
    at Array.map (<anonymous>)
    at index.tsx:162:18
    at handleOnChange (29062.2bb468e49a2eee40cded.js:1:2798047)
    at Object.eI (2541.b44fafd1acac6bd12301.js:1:797772)
    at eU (2541.b44fafd1acac6bd12301.js:1:797926)
    at 2541.b44fafd1acac6bd12301.js:1:816235
    at re (2541.b44fafd1acac6bd12301.js:1:816334)
    at rt (2541.b44fafd1acac6bd12301.js:1:816748)
```

This PR simply provides error handling on the URL resolution code, returning the favourited text as-is on failure.

This is obviously very niche behaviour, but would enable compatibility with any other plugin or utility that allows users to add arbitrary content to their gif picker.

### CODE REVIEW

**Comment by Vendicated** [comment]
oh god

what's the use case for storing random strings O.O

File: src/plugins/favGifSearch/index.tsx [hunk 1]
```diff
@@ -200,23 +200,28 @@ function SearchBar({ instance, SearchBarComponent }: { instance: Instance; Searc
...
-            return url.pathname;
-        case "hostandpath":
-            if (url.host === "media.discordapp.net" || url.host === "tenor.com")
-                return `${url.host} ${url.pathname.split("/").at(-1) ?? url.pathname}`;
-            return `${url.host} ${url.pathname}`;
-
-        default:
-            return "";
+    try {
+        const url = new URL(urlStr);
```
Comment (Vendicated) [inline src/plugins/favGifSearch/index.tsx:204]: could you please restructure this so that only the new URL() line is inside the try catch

**Comment by Ratismal** [comment]
Quick text snippets primarily. For example, I've had this in my picker since the picker was first rolled out.

https://github.com/Vendicated/Vencord/assets/1059362/a0797fb6-4fc0-4b0f-9f08-7325fb795d5c

**Comment by Vendicated** [comment]
thanks!

**Pull request merged**